### ISSUE TITLE
List of supported compilers, etc.

### ISSUE THREAD

**Issue by tim-lebedkov** [conan-io/conan#364]
I unfortunately have not found a list of supported compilers (e.g. the value "gcc"), platforms, etc.
I expected the list to be available here: http://docs.conan.io/en/latest/reference/conanfile.html#package-meta-information

BTW, what should be used for MinGW-w64 i686/SJLJ/POSIX threads?

**Comment by lasote** [comment]
You can find all the settings in the *~/.conan/settings.yml*
http://docs.conan.io/en/latest/reference/config_files.html?highlight=settings#settings-yml

You can edit this file, it's totally configurable. If you detect the need of a new setting please tell us to add it to the defaults.

**Comment by memsharded** [comment]
You should use: ``-s compiler=gcc -s compiler.version=4.9 -s arch=x86_64 -s compiler.libcxx=libstdc++`` Of course the version and libcxx of your compiler can be adjusted. With regards to SJLJ/POSIX threads, there are no default settings, but if the binaries generated with different mingw versions are indeed different, you can easily add something like ``exceptions: [SJLJ, SEH]`` and ``threads=[Posix, Win]`` to your settings.xml. 

It is true that  most packages in windows have binaries for MSVC, but MinGW support is not so common, so you might need to build them from sources, or even adapt some package recipe in order to provide MinGW support. Ask for help if needed.

**Comment by tim-lebedkov** [comment]
I think it is necessary to add this compiler setting for GCC/MinGW-w64:

`exceptions: [sjlj, seh, dwarf2]
`

The combination dwarf2/x86_64 is not supported.
As "C" does not use exceptions, the setting can be ignored there.

The POSIX/Windows distinction for the threads is not important to me. Here is some information about it: http://stackoverflow.com/a/30390278/126306

**Comment by GavinNL** [comment]
Hi,
I was having some issues compiling some stuff on Linux Mint 18 which I wasn't having a few weeks ago.  It kept complaining about my gcc compiler version being 5.4.  I didn't remember having this problems before. 

It turns out that Linux Mint 18 comes with gcc 5.3. Everthing was working fine, but one day mint decided to upgrade me to 5.4 during the regular system updates. That's when conan started complaining.  Anyways to fix it, I had to just add "5.4" to my settings.yml file in the gcc section, maybe this should be added by default?

**Comment by johnmcfarlane** [comment]
I'd assume so, yes. 

I just downloaded and installed Conan for the first time. I have Ubuntu 16.04 which updated its default GCC version from 5.3 to 5.4 recently. Here's my first attempt to use it:
```shell

It seems to be the first time you run conan
Auto detecting your dev setup to initialize conan.conf
Found gcc 5.4
Found clang 3.8
Default conan.conf settings
	os=Linux
	arch=x86_64
	compiler=gcc
	compiler.version=5.4
	compiler.libcxx=libstdc++
	build_type=Release
*** You can change them in ~/.conan/conan.conf ***
*** Or override with -s compiler='other' -s ...s***


ERROR: '5.4' is not a valid 'settings.compiler.version' value.
Possible values are ['4.4', '4.5', '4.6', '4.7', '4.8', '4.9', '5.1', '5.2', '5.3', '6.1']
```

I'm guessing this means that Conan doesn't work out of the box for most Ubuntu 16.04 users. This issue is closely related to [issue #26](https://github.com/conan-io/conan/issues/26).

**Comment by memsharded** [comment]
Hi Gavin, John,

Yes, didnt realize cause we are mostly using travis for testing, and Ubuntu 14 as reference distro. I added it to the default, checked that a migration will occur and submitted yesterday a PR with it: https://github.com/conan-io/conan/pull/380.

I should have reference this issue, sorry I forgot.

It will be released in next 0.11, most likely this week. In the meantime, yes, please, just add 5.4 to your /.conan/settings.yml and it will work.

Thanks very much for reporting and the feedback
